src/libostree/ostree-repo.h

```diff
@@ -761,6 +762,29 @@ typedef struct {
   gpointer unused_ptrs[7];
 } OstreeRepoCheckoutOptions;
 
+/**
+ * OstreeRepoCheckoutAtOptions:
+ *
+ * An extensible options structure controlling checkout.  Ensure that
+ * you have entirely zeroed the structure, then set just the desired
+ * options.  This is used by ostree_repo_checkout_at() which
+ * supercedes previous separate enumeration usage in
+ * ostree_repo_checkout_tree() and ostree_repo_checkout_tree_at().
+ */
+typedef struct {
+  OstreeRepoCheckoutMode mode;
+  OstreeRepoCheckoutOverwriteMode overwrite_mode;
+
+  gboolean enable_uncompressed_cache;
+  gboolean disable_fsync;
+  gboolean process_whiteouts;
+  gboolean no_copy_fallback;
+
+  const char *subpath;
+
+  OstreeRepoDevInoCache *devino_to_csum_cache;
+} OstreeRepoCheckoutAtOptions;
```

Comment:
Can you readd the padding bits:

```
  int unused_ints[6];
  gpointer unused_ptrs[7];
```

So that we can extend this in the future?